test/method_detection.test.js

```diff
@@ -1,37 +1,54 @@
+const fs = require('fs');
 const test = require('tap').test;
 
-var ast = require('../lib/ast.js');
+const ast = require('../lib/ast.js');
 
 test('test st method detection', function (t) {
-  var stPath = __dirname + '/fixtures/st/node_modules/st.js';
-  var methods = ['Mount.prototype.getPath'];
-  var line = 158;
-  var found = ast.findAllVulnerableFunctionsInScriptPath(
-    stPath, methods
+  const content = fs.readFileSync(__dirname + '/fixtures/st/node_modules/st.js');
+  const methods = ['Mount.prototype.getPath'];
+  const line = 158;
+  const found = ast.findAllVulnerableFunctionsInScript(
+    content, methods,
   );
   t.equal(found[methods[0]].line, line, 'Mount.prototype.getPath found');
   t.end();
 });
 
 test('test handlebars method detection', function (t) {
-  var stPath = __dirname + '/fixtures/handlebars/lib/handlebars/utils.js';
-  var methods = ['escapeExpression'];
-  var line = 63;
-  var found = ast.findAllVulnerableFunctionsInScriptPath(
-    stPath, methods
+  const content = fs.readFileSync(__dirname + '/fixtures/handlebars/lib/handlebars/utils.js');
+  const methods = ['escapeExpression'];
+  const line = 63;
+  const found = ast.findAllVulnerableFunctionsInScript(
+    content, methods,
   );
   t.equal(found[methods[0]].line, line, 'escapeExpression found');
   t.end();
 });
 
 test('test uglify-js method detection', function (t) {
-    var stPath = __dirname + '/fixtures/uglify-js/lib/parse.js';
-    var methods = ['parse_js_number'];
-    var line = 180;
-    var found = ast.findAllVulnerableFunctionsInScriptPath(
-      stPath, methods
-    );
-    t.equal(found[methods[0]].line, line, 'parse_js_number found');
-    t.end();
-  });
-  
+  const content = fs.readFileSync(__dirname + '/fixtures/uglify-js/lib/parse.js');
+  const methods = ['parse_js_number'];
+  const line = 180;
+  const found = ast.findAllVulnerableFunctionsInScript(
+    content, methods,
+  );
+  t.equal(found[methods[0]].line, line, 'parse_js_number found');
+  t.end();
+});
+
+test('test export = { f() {} } method detection', function (t) {
+  const contents = `
+module.exports = {
+  foo() {},
+  bar() {},
+};
+`;
+  const methods = ['module.exports.foo', 'module.exports.bar'];
```

Comment:
Is this the way they appear in functions-to-track-runtime.json? I think they would appear just as 'foo' and 'bar' (like 'publish'). If so - you should take it into consideration in the inspectNode function

Reply:
We're defining a new format, it's up to us to come up with an unambiguous way to name methods. Perhaps we're not doing a good job of this elsewhere.


These two functions, both named `foo`, are distinct. We already support and use this `module.exports.foo` syntax for this second function:
```
function foo() {}
module.exports.foo = function() => {}
```

And that code is, as far as I'm aware, equivalent to:
```
function foo() {}
module.exports = { function foo() {} };
```

... so generating the same name seems reasonable.

---

There's going to be cases where we'd need full flow/halting analysis to find out what actually happened, which is not reasonable right now:
```
module.exports = { function foo() { return 1; } };
if (rand()) {
  module.exports.foo = function() { return 2; };
}
```